File: YYProject/NextViewController/HUDViewController.m
//
//  HUDViewController.m
//  YYProject
//
//  Created by 于优 on 2018/12/5.
//  Copyright © 2018 SuperYu. All rights reserved.
//

#import "HUDViewController.h"
#import "YYProgressHUD.h"
#import "UIView+Extension.h"

@interface HUDViewController ()

/** 数据源 */
@property (nonatomic, strong) NSArray<NSString *> *dataArray;
/**  */
@property (nonatomic, strong) UIButton *saveBtn;

@end

@implementation HUDViewController

static NSString *const toolCell = @"toolCell";

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self createView];
}

- (void)createView {
    
    self.navView.title = @"HUD";
    self.navView.seperateColor = kBlackColor;
    
    [self.view addSubview:self.tableView];
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:toolCell];
    [self.tableView setSeparatorStyle:UITableViewCellSeparatorStyleSingleLine];
    
    [self.tableView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self.navView.mas_bottom).offset(0);
        make.left.bottom.right.equalTo(self.view).offset(0);
    }];
}

#pragma mark - TabeView

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    
    return self.dataArray.count;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    return 60.f;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:toolCell forIndexPath:indexPath];
    cell.textLabel.text = self.dataArray[indexPath.row];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    
    switch (indexPath.row) {
        case 0:
            [YYProgressHUD showLoadingStyle:YYHUDLoadingStyleDeterminate];
            break;
        case 1:
            [YYProgressHUD showLoadingStyle:YYHUDLoadingStyleDeterminateHorizontalBar];
            break;
        case 2:
            [YYProgressHUD showLoadingStyle:YYHUDLoadingStyleAnnularDeterminate];
            break;
        case 3:
            [YYProgressHUD showPlainText:@"文字提醒" view:self.view];
            break;
        case 4:
            [YYProgressHUD showSuccess:@"已关注" toView:self.view];
            break;
        case 5:
            [YYProgressHUD showError:@"操作失败" toView:self.view];
            break;
        case 6:
            [YYProgressHUD showIcon:[UIImage imageNamed:@"contacts"] message:@"自定义icon"];
            break;
        case 7:
            [YYProgressHUD showCustomView:self.saveBtn hideAfterDelay:1];
            break;
            
        default:
            break;
    }
}

#pragma mark - setter & getter

- (NSArray<NSString *> *)dataArray {
    if (!_dataArray) {
        _dataArray = @[@"LoadingStyleDeterminate", @"LoadingStyleHorizontalBar", @"LoadingStyleAnnularDeterminate", @"showPlainText", @"showSuccess", @"showError", @"showIcon", @"showCustomView"];
    }
    return _dataArray;
}

- (UIButton *)saveBtn {
    if (!_saveBtn) {
        _saveBtn = [UIButton buttonWithType:UIButtonTypeCustom];
        [_saveBtn setTitle:@"确定" forState:UIControlStateNormal];
        [_saveBtn setTitleColor:kWhiteColor forState:UIControlStateNormal];
        _saveBtn.titleLabel.font = FONT_SIZE(12);
        _saveBtn.backgroundColor = kFontColor_Theme;
        [_saveBtn setLayerRoundedRect:10];
    }
    return _saveBtn;
}

@end

File: YYProject/NextViewController/HUDViewController.swift
//
//  HUDViewController.swift
//  YYProject
//
//  Created by 于优 on 2018/12/11.
//  Copyright © 2018 SuperYu. All rights reserved.
//

import UIKit
import SnapKit

class HUDViewController: YYBaseViewController {
    
    let dataArray:[String] = ["LoadingStyleDeterminate", "LoadingStyleHorizontalBar", "LoadingStyleAnnularDeterminate", "showPlainText", "showSuccess", "showError", "showIcon", "showCustomView"]
    
    let toolCell = "toolCell"
    
    

    override func viewDidLoad() {
        super.viewDidLoad()
        
        createView()
    }
    
    func createView() {
        navView.title = "HUD"
        navView.seperateColor = UIColor.lightGray
        
        self.view.addSubview(tableView)
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: toolCell)
        tableView.separatorStyle = .singleLine
        
//        tableView.snp.makeConstraints { (make) -> Void in
//            make.top.equalTo(navView.snp.bottom).offset(0)
//            make.left.bottom.right.equalTo(view)
//        }
        
        tableView.snp.makeConstraints { (ConstraintMaker) in
            ConstraintMaker.top.equalTo(navView.snp.bottom).offset(0)
            ConstraintMaker.left.bottom.right.equalTo(view)
        }
    }
    
    // MARK:-TableView
    
   override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

   override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return dataArray.count
    }
    
    override func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
        return 60
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        
        return 60
    }
    
   override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: toolCell, for: indexPath)
        cell.textLabel?.text = dataArray[indexPath.row]
        cell.accessoryType = .disclosureIndicator
        return cell
    }
    
   override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        switch indexPath.row {
        case 0:
            YYProgressHUD.showLoading(YYHUDLoadingStyleDeterminate, message: "加载中...")
        case 1:
            YYProgressHUD.showLoading(YYHUDLoadingStyleDeterminateHorizontalBar, message: "加载中...")
        case 2:
            YYProgressHUD.showLoading(YYHUDLoadingStyleAnnularDeterminate, message: "加载中...")
        case 3:
            YYProgressHUD.showPlainText("文字提醒", view: view)
        case 4:
            YYProgressHUD.showSuccess("已关注", to: view)
        case 5:
            YYProgressHUD.showError("操作失败", to: view)
        case 6:
            let image = UIImage(named: "contacts")
            YYProgressHUD.showIcon(image!, message: "自定义icon")
        case 7:
            let btn = UIButton(type: UIButton.ButtonType.custom)
            btn.backgroundColor = UIColor.darkGray
            btn.layer.cornerRadius = 6.0
            btn.clipsToBounds = true
            btn.setTitle("确定", for:UIControl.State.normal)
            btn.titleLabel?.font = UIFont.init(name: "PingFangSC-Regular", size: 12)
            btn.setTitleColor(UIColor.white, for: .normal)
            YYProgressHUD.showCustomView(btn, hideAfterDelay: 2)
        default:
            break
        }
    }
}

